__tests__/utilities/test-utils.js

```diff
@@ -0,0 +1,80 @@
+const fs = require('fs');
+const _ = require('lodash');
+const path = require("path");
+const RudderCDK = require("rudder-transformer-cdk");
+
+function formTestParams(dest, transformAt) {
+	const version = "v0";
+  
+  const transformer = require(
+    path.resolve(__dirname + `../../../${version}/destinations/${dest}/transform`)
+  );
+  
+  //for router test
+  let trCat = '';
+  if (transformAt === 'router') {
+    trCat += 'router_'
+  }
+  const inputDataFile = fs.readFileSync(
+    path.resolve(__dirname, `../data/${dest}_${trCat}input.json`)
+  );
+  const outputDataFile = fs.readFileSync(
+    path.resolve(__dirname, `../data/${dest}_${trCat}output.json`)
+  );
+  const inputData = JSON.parse(inputDataFile);
+  const expected = JSON.parse(outputDataFile);
+	const cdkDest = ['variance'].filter(({ name }) => name === dest)[0];
+	return {
+		input: inputData,
+		expected,
+		transformer,
+		iscdkDest: _.isEmpty(cdkDest)
+	};
+}
+
+function executeTransformationTest(dest, transformAt) {
+  const testParams = formTestParams(dest, transformAt);
+  const { iscdkDest, transformer, expected, input } = testParams;
+  
+  const basePath = path.resolve(__dirname, "../../cdk");
+  const factory = new RudderCDK.ConfigFactory(basePath);
+
+  describe(`${dest} Tests`, () => {
+    it(`${dest} - ${transformAt} tests`, async () => {
+      let actualData = [];
+      if (transformAt === 'processor') {
+        actualData = await Promise.all(
+          input.map(async inp => {
+            try {
+              if (!iscdkDest) {
+                return await transformer.process(inp)
+              }
+              return await RudderCDK.Executor.executeStages(
+                inp,
+                factory.getConfig(dest)
+              )
+              expect(actualData).toEqual(expected);
+            } catch (error) {
+              console.log(error.message)
+              return error.message
+            }
+          })
+        );
+        actualData.map((actData, index) => {
+          console.log("Testing Payload no ",index)
+          if (expected[index].error) {
+            expect(actData).toEqual(expected[index].error);
+          } else {
+            expect(actData).toEqual(expected[index])
+          }
+        });
+      } else {
+        actualData = await transformer.processRouterDest(input);
+        expect(actualData).toEqual(expected);
```

Comment:
Why isn't any elaborate checking made for router tests? Ideally, except for the API to be called on the transformer, everything should be identical.

Reply:
Sorry I don't understand this. What do you mean by elaborate checking in this context ?

Reply:
There should not be any difference b/w the way in which processor and router tests are handled except for the API (`process` vs `processRouterDest`).

Reply:
`process` takes input as a single event and returns an array of transformed payload
`processRouterDest` takes input as an array of events and returns an array of transformed payload

The transformed payload for both of them are different, it'll be difficult to bring them under same roof.

Reply:
But let me check if such kind of thing is possible, we can leverage it

Reply:
No, I meant something like this.

```
// Here "input" is an array of events for processor tests and
// array of array of events for router tests.
input.forEach(tcInput => {
  it(`test name`, async () => {
    try {
      if (transformAt == "processor") {
        actualData = transformer.process(tcInput);
      } else {
        actualData = tranformer.processRouterDest(tcInput);
      }
      // Compare actual and expected data
    }
    catch {
      // Force fail the test case if the expected exception is not raised
    }
  )};
}
```

Reply:
Changed this. Can you once check and let me know if there are anymore concerns ?